### PR TITLE
feat: add custom tags to created ECR repositories

### PR DESCRIPTION
**Pull request by darkweaver87**
This PR adds the ability to configure custom tags for created repositories.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -64,7 +63,7 @@ A mutating webhook for Kubernetes, pointing the images to a new location.`,
 		//metricsRec := metrics.NewPrometheus(promReg)
 		log.Trace().Interface("config", cfg).Msg("config")
 
-		rClient, err := registry.NewECRClient(cfg.Target.AWS.Region, cfg.Target.AWS.EcrDomain())
+		rClient, err := registry.NewECRClient(cfg.Target.AWS.Region, cfg.Target.AWS.EcrDomain(), cfg.RepositoryCustomTags)
```
Comment (estahn) [inline cmd/root.go:66]: How about moving this to a separate method, e.g. `setRepositoryCustomTags`?

Comment (darkweaver87) [reply cmd/root.go:66]: Hello,

Thanks for your comment. I'll do :-)

Rémi

**Comment by project0** [comment]
@estahn is something missing here?

File: Dockerfile [hunk 2]
```diff
@@ -8,7 +8,8 @@
 FROM alpine:3.16.3
 RUN ["apk", "add", "--no-cache", "--repository=http://dl-cdn.alpinelinux.org/alpine/edge/community", "skopeo>=1.2.0"]
 
-COPY k8s-image-swapper /
+ARG TARGETARCH
+COPY k8s-image-swapper-$TARGETARCH /k8s-image-swapper
```
Comment (estahn) [inline Dockerfile:12]: @darkweaver87 Why is this change needed?

**Review by estahn — COMMENTED** [review]
Apologies it took me so long to come back to you. Looks good 🚀  , see comment re Docker file.

File: Dockerfile (hunk 2 above)
Comment (darkweaver87) [reply Dockerfile]: Actually it doesn't seem to be needed with go-releaser :-)

**Comment by darkweaver87** [comment]
Thanks for the review :-)

**Comment by estahn** [comment]
It appears like I made provisions for this use-case back in the day but didn't actually implement it:

https://github.com/estahn/k8s-image-swapper/blob/main/.k8s-image-swapper.yml#L45-L48

Do you think you can adjust the changes to use the configuration from `target.aws.ecrOptions.tags` instead?

**Comment by darkweaver87** [comment]
Is it ok this way ?

**Comment by estahn** [comment]
Looking good so far. Will give it another review a bit later or tomorrow morning.

File: docs/configuration.md [hunk 3]
```diff
@@ -138,3 +138,19 @@ The AWS Account ID and Region is primarily used to construct the ECR domain `[AC
...
+
+!!! example
+    ```yaml
+    target:
+      type: aws
+        aws:
+          ecrOptions:
+            tags:
+              - name: cluster
+                value: myCluster
```
Comment (estahn) [inline docs/configuration.md:155]: This needs to go under `target`.
```suggestion
      aws:
        ecrOptions:
          tags:
            - name: cluster
              value: myCluster
```

File: docs/configuration.md [hunk 4]
```diff
@@ -138,3 +138,19 @@ The AWS Account ID and Region is primarily used to construct the ECR domain `[AC
         accountId: 123456789
         region: ap-southeast-2
     ```
+
+### RepositoryTags
```
Comment (estahn) [inline docs/configuration.md:142]: I would prefer to have the documentation mirror the configuration options for easier navigation. Thoughts?

```suggestion
### ECR Options

#### Tags
```

**Review by estahn — CHANGES_REQUESTED** [review]
Looking really good. Thanks so much 👍🏼 

See comments for some minor docu changes and we can merge after.

Again, thanks heaps for getting this done.

**Comment by darkweaver87** [comment]
All done :-)

**Review by estahn — APPROVED** [review]
